mailbox/event/event-rabbitmq/src/main/java/org/apache/james/mailbox/events/RabbitMQEventBus.java

```diff
@@ -44,29 +45,45 @@
     private static final boolean DURABLE = true;
     private static final String DIRECT_EXCHANGE = "direct";
 
-    private final Sender sender;
-    private final GroupRegistrationHandler groupRegistrationHandler;
-    private final KeyRegistrationHandler keyRegistrationHandler;
-    private final EventDispatcher eventDispatcher;
+    private final Mono<Connection> connectionMono;
+    private final EventSerializer eventSerializer;
+    private final RoutingKeyConverter routingKeyConverter;
+
+    private GroupRegistrationHandler groupRegistrationHandler;
+    private KeyRegistrationHandler keyRegistrationHandler;
+    private EventDispatcher eventDispatcher;
+    private AtomicBoolean isRunning;
+    private Sender sender;
 
     RabbitMQEventBus(RabbitMQConnectionFactory rabbitMQConnectionFactory, EventSerializer eventSerializer, RoutingKeyConverter routingKeyConverter) {
-        Mono<Connection> connectionMono = Mono.fromSupplier(rabbitMQConnectionFactory::create).cache();
-        this.sender = RabbitFlux.createSender(new SenderOptions().connectionMono(connectionMono));
-        this.groupRegistrationHandler = new GroupRegistrationHandler(eventSerializer, sender, connectionMono);
-        this.eventDispatcher = new EventDispatcher(eventSerializer, sender);
-        this.keyRegistrationHandler = new KeyRegistrationHandler(eventSerializer, sender, connectionMono, routingKeyConverter);
+        this.connectionMono = Mono.fromSupplier(rabbitMQConnectionFactory::create).cache();
+        this.eventSerializer = eventSerializer;
+        this.routingKeyConverter = routingKeyConverter;
+
     }
 
     public void start() {
-        eventDispatcher.start();
-        keyRegistrationHandler.start();
+        if (isRunning == null || !isRunning.get()) {
+            sender = RabbitFlux.createSender(new SenderOptions().connectionMono(connectionMono));
```

Comment:
once `sender` is close, there is no way to reuse it but only re create it. This issue leads to re create new a set of objects `groupRegistrationHandler`, `eventDispatcher`, `keyRegistrationHandler` because they need `sender` also. 

P/S: For supporting multiple start & stop continuously.